#10226: Added or refactored BH SFPU functions

ncvetkovicTT opened the pull request:
### Ticket
#10226 

### Problem description
Some functions are not added/enabled for BH.

### What's changed
Within the scope of this PR, the following changes are introduced:

1. SFPU mask - added _int version
2. SFPU reciprocal - modified to use submodule LLKs
3. SFPU sqrt - modified to use submodule LLKs
4. SFPU trunc OP - enabled for BH

### Checklist
- [x] Post commit CI passes - [#15330](https://github.com/tenstorrent/tt-metal/actions/runs/10788610231)
- [x] Blackhole Post commit (if applicable) - [#584](https://github.com/tenstorrent/tt-metal/actions/runs/10790977602)
- [ ] Model regression CI testing passes (if applicable)
- [x] New/Existing tests provide coverage for changes

rtawfik01 commented:
Since Blackhole only has SD/FD C++ tests on pipelines, can you also run this `pytest -svv tests/tt_eager/python_api_testing/sweep_tests/pytests/tt_dnn/test_eltwise_unary.py` and make sure it still passes since it has not been added to CI yet?

rtawfik01 commented:
Also how has the `reshuffle_rows` been tested?

ncvetkovicTT committed:
#10226: Added or refactored BH SFPU functions

1. SFPU mask - added _int version
2. SFPU reciprocal - modified to use submodule LLKs
3. SFPU sqrt - modified to use submodule LLKs
4. SFPU trunc OP - enabled for BH

ncvetkovicTT commented:
> Since Blackhole only has SD/FD C++ tests on pipelines, can you also run this `pytest -svv tests/tt_eager/python_api_testing/sweep_tests/pytests/tt_dnn/test_eltwise_unary.py` and make sure it still passes since it has not been added to CI yet?

Tested, pass ✅

ncvetkovicTT commented:
> Also how has the `reshuffle_rows` been tested?

No, the work will be continued as a part of [#11816](https://github.com/tenstorrent/tt-metal/tree/ncvetkovic/11816-reshuffle_rows_support)

abhullar-tt reviewed (approved):
I believe this should unblock some of the ops that are failing in #12349

eyonland reviewed (changes requested):
I'd like to request that some sample of the uses cases of the sweep tests mentioned by @rtawfik01  be added to tests/ttnn/unit_tests

abhullar-tt commented:
> I'd like to request that some sample of the uses cases of the sweep tests mentioned by @rtawfik01 be added to tests/ttnn/unit_tests

We are working to enable the fast-dispatch-post-commit workflow on BH CI which should cover these cases, there is a PR up where we are disabling the current failing tests on BH (see https://github.com/tenstorrent/tt-metal/tree/mchiou/11919-enable-ttnntteager-on-bh-ci)

eyonland reviewed (approved):
Based on conversation with @abhullar-tt it sounds like we have a parallel effort and we will get unit tests turned on eventually.
